keras_cv/layers/preprocessing/with_segmentation_masks_test.py

```diff
@@ -0,0 +1,101 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from absl.testing import parameterized
+
+from keras_cv.layers import preprocessing
+
+TEST_CONFIGURATIONS = [
+    ("AutoContrast", preprocessing.AutoContrast, {"value_range": (0, 255)}),
+    ("ChannelShuffle", preprocessing.ChannelShuffle, {}),
+    ("Equalization", preprocessing.Equalization, {"value_range": (0, 255)}),
+    ("Grayscale", preprocessing.Grayscale, {}),
+    ("GridMask", preprocessing.GridMask, {}),
+    (
+        "Posterization",
+        preprocessing.Posterization,
+        {"bits": 3, "value_range": (0, 255)},
+    ),
+    (
+        "RandomColorDegeneration",
+        preprocessing.RandomColorDegeneration,
+        {"factor": 0.5},
+    ),
+    (
+        "RandomHue",
+        preprocessing.RandomHue,
+        {"factor": 0.5, "value_range": (0, 255)},
+    ),
+    (
+        "RandomChannelShift",
+        preprocessing.RandomChannelShift,
+        {"value_range": (0, 255), "factor": 0.5},
+    ),
+    (
+        "RandomColorJitter",
+        preprocessing.RandomColorJitter,
+        {
+            "value_range": (0, 255),
+            "brightness_factor": (-0.2, 0.5),
+            "contrast_factor": (0.5, 0.9),
+            "saturation_factor": (0.5, 0.9),
+            "hue_factor": (0.5, 0.9),
+            "seed": 1,
+        },
+    ),
+    (
+        "RandomGaussianBlur",
+        preprocessing.RandomGaussianBlur,
+        {"kernel_size": 3, "factor": (0.0, 3.0)},
+    ),
+    ("RandomJpegQuality", preprocessing.RandomJpegQuality, {"factor": (75, 100)}),
+    ("RandomSaturation", preprocessing.RandomSaturation, {"factor": 0.5}),
+    (
+        "RandomSharpness",
+        preprocessing.RandomSharpness,
+        {"factor": 0.5, "value_range": (0, 255)},
+    ),
+    ("Solarization", preprocessing.Solarization, {"value_range": (0, 255)}),
+]
+
+
+class WithSegmentationMasksTest(tf.test.TestCase, parameterized.TestCase):
+    @parameterized.named_parameters(*TEST_CONFIGURATIONS)
+    def test_can_run_with_segmentation_masks(self, layer_cls, init_args):
+        layer = layer_cls(**init_args)
+
+        img = tf.random.uniform(
+            shape=(3, 512, 512, 3), minval=0, maxval=1, dtype=tf.float32
+        )
+        segmentation_masks = tf.random.uniform(
+            shape=(3, 512, 512, 3), minval=0, maxval=1, dtype=tf.int32
+        )
+
+        inputs = {"images": img, "segmentation_masks": segmentation_masks}
+        _ = layer(inputs)
```

Comment:
We should validate that the output mask is not changed (no op).

Reply:
Done, although this won't always be true (once we add non-noop KPLs for seg masks). I've left a comment about this, and we can add a parameter for this to the tests in the future.